### ISSUE TITLE
Event history

### ISSUE THREAD

**Issue by alesribic73** [Ylianst/MeshCentral#880]
Hi, where can i get more events than 1000. I need events for a year or even more because of GDPR AuditLog requests.

**Comment by Ylianst** [comment]
Agreed, I need to add more sophisticated filters on pages that display events. On a related topic, by default MeshCentral will remove records after a given time, you can set this in the "Settings" section of the config.json, here are the defaults in seconds:

```
    "DbExpire": {
      "events": 1728000,
      "powerevents": 864000,
      "statsevents": 2592000
    }
```

**Comment by alesribic73** [comment]
I have set per your instruction.  Will drop down menu show now more results, or will be still limited to 1000?

**Comment by Ylianst** [comment]
The settings above are the default. They are the settings used if you don't provide any values. If you want to keep events in the database for longer, you must increase these values. Also, 1000 is still the maximum you can view until it's improved. Of course, if using MongoDB, you can always query the database directly.

**Comment by alesribic73** [comment]
Ok, thanks. So events are in the database, but it can not be shown, until improvement.

Thanks

**Comment by Ylianst** [comment]
Correct. If you want to keep events for a year or more, make sure to change the values above to much bigger values, put the number of seconds in a year for starters.

**Comment by desarrollocdb** [comment]
@Ylianst It would be interesting to be able to export all the records in CSV or JSON, that is, the "show" tab has an option that says all and that they can only be exported.

**Comment by si458** [comment]
you can now export events to csv or json
![image](https://github.com/Ylianst/MeshCentral/assets/765314/772a2f4c-c265-464d-ba1e-ffb39fbc836b)
the is also a no limit option to list all events, but this might be very huge and very large!
we still need to add a time filter option maybe?

**Comment by si458** [comment]
you can use `meshctrl.js` to export the data as json or raw or csv and then filter out the data yourself
`node meshctrl.js ListEvents --url wss://localhost --loginuser youruser--loginpass yourpass --json`
```
simon@testmc:~/meshcentral/MeshCentral$ node meshctrl.js help ListEvents
List server events optionally filtered by user or device. Example usage:

  MeshCtrl ListEvents 

Optional arguments:

  --userid [name]        - User account identifier.
  --id [deviceid]        - The device identifier.
  --limit [number]       - Maximum number of events to list.
  --raw                  - Output raw data in JSON format.
  --json                 - Give results in JSON format.
```
